raster/r.to.rast3elev/main.c

```diff
@@ -408,7 +408,7 @@ int main(int argc, char *argv[])
 
     /*Set the upper value */
     if (param.upper->answer) {
-        if (sscanf(param.upper->answer, "%lf", &db.upper))
+        if (sscanf(param.upper->answer, "%lf", &db.upper) == 1)
```

Comment:
`G_parser()` does all sanity checks to ensure it returns a non-NULL only if any value is provided to an option. An empty value will return a NULL, not `"\0"`, so it shoud be enough to check for `NULL`.

https://github.com/OSGeo/grass/blob/87d2d42d3b26c01d6de08ae8845618c9918a20cc/lib/gis/parser.c#L1180-L1182

Reply:
kept in as it seems good practice as stated in an earlier comment